qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -280,6 +280,21 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
         return self._run(dag, tracker)
 
     def _run(self, dag: DAGCircuit, tracker: QubitTracker) -> DAGCircuit:
+        # Check if HighLevelSynthesis can be skipped.
+        all_skipped = True
+        for node in dag.topological_op_nodes():
+            qubits = tuple(dag.find_bit(q).index for q in node.qargs)
+            if not (
```

Comment:
Would it make sense to cache this result so we can just look it up in the second iteration over the DAG?

Reply:
You mean hashing whether a particular node can be skipped? I briefly tried this, but in the main use-case, where only a single pass is required, this extra hashing makes the runtime a bit slower.